src/platform_impl/ios/window.rs

```diff
@@ -493,86 +497,104 @@ impl Inner {
         unsafe {
             let status_bar_hidden = if hidden { YES } else { NO };
             let () = msg_send![
-                self.view_controller,
+                *self.view_controller,
                 setPrefersStatusBarHidden: status_bar_hidden
             ];
         }
     }
 }
 
 impl Inner {
-    // requires main thread
+    // executed on main thread
     unsafe fn screen_frame(&self) -> CGRect {
-        self.to_screen_space(msg_send![self.window, bounds])
+        let window = self.window;
+        Queue::main().exec_sync(move || {
```

Comment:
> The way our event loop works on iOS is that we call run and process all the windowing events on another thread (we send them over an mpsc channel). However, on iOS a lot of things need to happen on the main thread as documented / asserted in winit.

It seems like this would deadlock for anyone calling these methods who is *already* on the main thread. Is it reasonable to say that users of this lib can't call these methods from the main thread. That is surprising to me.

A better solution for your situation might be to hoist these dispatches to your client code.

Or am I misunderstanding?